### PR TITLE
Iox #571 replace timespec by duration in semaphore

### LINKED ISSUE #571: Replace timespec by Duration in semaphore
## Brief feature description

Replace timespec struct by ``iox::units::Duration`` in ``timedWait()`` method of ``iox::posix::Semaphore``.

### PR DESCRIPTION
**Pull request by FerdinandSpitzschnueffler**
## Pre-Review Checklist for the PR Author

1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## Post-review Checklist for the Eclipse Committer

1. [x] All checkboxes in the PR checklist are checked or crossed out
1. [x] Merge

## References

- Closes #571

### CODE REVIEW

File: iceoryx_utils/test/moduletests/test_semaphore_module.cpp [hunk 1]
```diff
@@ -323,10 +316,10 @@ TIMING_TEST_P(Semaphore_test, TimedWaitWithTimeout, Repeat(3), [&] {
     std::atomic_bool timedWaitFinish{false};
 
     std::thread t([&] {
-        auto ts = Duration::fromNanoseconds(TIMING_TEST_TIMEOUT).timespec(TimeSpecReference::Epoch);
+        auto ts = Duration::fromNanoseconds(TIMING_TEST_TIMEOUT);
```
Comment (elBoberido) [inline iceoryx_utils/test/moduletests/test_semaphore_module.cpp:319]: nitpicking, this isn't a `ts` anymore ;)

File: iceoryx_utils/test/moduletests/test_semaphore_module.cpp [hunk 2]
```diff
@@ -349,10 +342,10 @@ TIMING_TEST_P(Semaphore_test, TimedWaitWithoutTimeout, Repeat(3), [&] {
...
```
Comment (elBoberido) [inline iceoryx_utils/test/moduletests/test_semaphore_module.cpp:345]: same here

**Review by elBoberido — COMMENTED** [review]
looks good, except the nitpicking with the variable names

File: iceoryx_utils/test/moduletests/test_semaphore_module.cpp (hunk 1 above)
Comment (FerdinandSpitzschnueffler) [reply iceoryx_utils/test/moduletests/test_semaphore_module.cpp:319]: upsi

**Review by elBoberido — APPROVED** [review]

**Review by elfenpiff — APPROVED** [review]

**Pull request merged**